src/server/pfs/drive/driver.go

```diff
@@ -1071,6 +1072,11 @@ func (d *driver) inspectFile(file *pfs.File, filterShard *pfs.Shard, shard uint6
 	if fileInfo.FileType == pfs.FileType_FILE_TYPE_NONE {
 		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
 	}
+	// We return NotFound if it's a regular file but its blocks do not match the
+	// given filter shard
+	if fileInfo.FileType == pfs.FileType_FILE_TYPE_REGULAR && len(blockRefs) == 0 && (filterShard != nil && filterShard.BlockNumber > 1) {
+		return nil, nil, pfsserver.NewErrFileNotFound(file.Path, file.Commit.Repo.Name, file.Commit.ID)
+	}
```

Comment:
Interesting, I think that suggests we have a different bug. It's sort of incidental that filterShard is always `nil` when we have an open commit isn't it? Here's a question, did this PR make it so that empty writes will result in no blocks being appended? Until recently empty writes would lead to a single empty block which was probably shielding us from this behavior before.